BDArmory/FX/FuelLeakFX.cs

```diff
@@ -172,14 +191,18 @@ void FixedUpdate()
                         if (mp.amount >= 0)
                         {
                             //part.RequestResource("MonoPropellant", ((double)drainRate * Mathf.Clamp((float)mp.amount, 40, 400) / Mathf.Clamp((float)mp.maxAmount, 400, (float)mp.maxAmount)) * Time.deltaTime);
-                            mp.amount -= ((double)drainRate * Mathf.Clamp((float)mp.amount, 40, 400) / Mathf.Clamp((float)mp.maxAmount, 400, (float)mp.maxAmount)) * Time.fixedDeltaTime;
+                            double amount = ((double)drainRate * Mathf.Clamp((float)mp.amount, 40, 400) / Mathf.Clamp((float)mp.maxAmount, 400, (float)mp.maxAmount)) * Time.fixedDeltaTime;
+                            mp.amount -= amount;
                             mp.amount = Mathf.Clamp((float)mp.amount, 0, (float)mp.maxAmount);
                             fuelLeft++;
+                            float density = 4f; // PartResourceLibrary.Instance.GetDefinition("MonoPropellant").density; // 5 kg/l for fuel/oxidizer, 4 kg/l for monopropellant
```

Comment:
The resource info is already available via `mp.info` (e.g., `mp.info.density`), so there's no need to hard-code it or look it up again in the `PartResourceLibrary`.